### PR TITLE
updating to vuetify 2 and much more

### PR DESCRIPTION
**Pull request by markim**
## Upgrade to Vuetify 2

Changes done:

### Architecture

- upgraded to vuetify 2
- upgraded all packages to latest versions (as of 3/27/2020)
- added vscode debug launch file (start vue app `npm run serve`, then f5 in vscode to debug)
- translations for error messages of inputs in "messages" root node of translation files
- v-text-field uses new `clear-icon=` and `clearable`, so cypress tests were updated to clear input manually until the clear icon can be clicked through cypress, along with adding some id's to some elements to more easily select from cypress
- added tests for light / dark
- prettier upgrade makes code es6 compliant (mainly: https://eslint.org/docs/rules/arrow-parens and a slightly diffferent line size from what I could tell)

### Website

- added light / dark mode to side bar and nav bar
	- switch hides on navbar for mobile, goes to sidebar
- fixed toolbar and other small vuetify parts that were broken
- small UI changes to give slightly better spacing in some areas
- update v-tables on admin cities and users vuetify datatable properties for proper item sync/pagination
- fixed some colors for light/dark changing on light mode
- theme stays in localStorage and persists on refresh

---

#### Notes

- I'm not sure if how I added the validation providers to each page with forms is correct. Seems simple enough to implement where needed, but maybe there's a better way? Please test heavily.
- I believe we could be using this method to fire off the submit as well: https://logaretm.github.io/vee-validate/api/validate.html
- vee-validate got rid of the 'url' rule in v3
	- used regex from this comment to create rule to satisfy code for github/twitter url's
	- https://github.com/logaretm/vee-validate/issues/2191#issuecomment-548425643

### CODE REVIEW

File: src/components/AdminCities.vue [hunk 1]
```diff
@@ -257,6 +284,7 @@ export default {
     },
     editItem(item) {
       this.editedItem = Object.assign({}, item)
+      console.log(item)
```
Comment (davellanedam) [inline src/components/AdminCities.vue]:
```suggestion
```
This is safe to remove

File: src/components/Home.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <v-container fluid>
     <v-layout row wrap>
-      <div class="text-xs-center">
+      <div align="center">
```
Comment (davellanedam) [inline src/components/Home.vue]: This attribute is not supported with HTML5

File: src/components/core/Footer.vue [hunk 3]
```diff
@@ -1,13 +1,11 @@
 <template>
-  <v-footer height="60" class="mt-5 justify-center text-xs-center d-block">
-    <div class="mt-2">
+  <v-footer height="60" align="center" class="d-block">
```
Comment (davellanedam) [inline src/components/core/Footer.vue]: This attribute `align` is not supported with HTML5

**Comment by davellanedam** [comment]
This is amazing markim! I ran tests and everything is passing. This looks really god!

**Comment by markim** [comment]
Thanks @davellanedam I'm glad to help :)

Commented out the console log, registered the new validation at a higher level instead of each component, and then fixed the alignments to use vuetify 2's classes "text-center" instead of "text-xs-center" and right and left and so on. IDK why I decided to use that old HTML tag, hah.

-Mark

**Comment by markim** [comment]
Hey @davellanedam I have made the requested changed. Please let me know if there is something else I can fix, or if this is good to go :)

I also bumped the version to v5.0.0, just a heads up!

Thanks

**Comment by davellanedam** [comment]
`package'lock.json` is funky. I'll fix it in `development` branch. Will merge this, thank you so much for your work @markim 👍 👍 👍

**Comment by davellanedam** [comment]
Hey @markim just found a small bug. When in mobile and logged in the admin menu section is still being rendered on top. Would you mind to create a PR for it? Thanks man! Your changes are in master now, and also in the demo site.

**Comment by markim** [comment]
Oops! Yep, will do. Thanks for the heads up, I'll see if I can fix this now.

**Comment by davellanedam** [comment]
@markim It's also weird I'm getting -Exit code 137- when doing nom run build in my web server. Locally builds ok, but on the server now fails. I had to build locally and upload demo manually. I'm debugging and looks like an out of memory error. At least what I found, but my server only has 25% of used memory. It's kind of weird. Will let you know more, if I'm able to find more...

**Comment by markim** [comment]
Hey @davellanedam  I made the changes and will create the pull request now.

That is weird. Seems like maybe the docker service running on the server might not have been configured with enough memory.
https://github.com/moby/moby/issues/21083

So maybe the server has 4gb of ram for example, but docker is set to start and use only 1gb max.

Let me know! Making merge request now.

**Comment by davellanedam** [comment]
@markim I'm not using Docker, hosting is on DigitalOcean, an Ubuntu 18.04.3 droplet. Started to happen with this update.

**Comment by davellanedam** [comment]
@markim Issue persists on 5.0.2. Will keep investigating on this.

**Comment by markim** [comment]
Ah gotcha. 
What are the server specs?
Can you describe the steps you used on a fresh VM? I can try and replicate the build. I know the front ends usually use more memory, but that's quite a hit from a vuetify upgrade.

Sounds great, thanks again for the project :)

**Comment by davellanedam** [comment]
@markim Just released 5.0.3. FIX: Button for saving on changing password was not rendering correctly. https://github.com/davellanedam/vue-skeleton-mvp/pull/233/files

**Comment by davellanedam** [comment]
@markim It's a basic VM (1 GB Memory / 25 GB SSD Disk)

**Comment by davellanedam** [comment]
@markim Thank you man for helping me out with this!

**Comment by davellanedam** [comment]
Steps are pretty straightforward:

1. Pull latest from master
2. npm I
3. npm run build
4. cp -R /WHATEVER_PATH/vue-skeleton-mvp-src/dist/* /WHATEVER_HTML_PUBLIC/vue

**Comment by markim** [comment]
What version of Node is installed on the server? I'll give it a shot now

**Comment by davellanedam** [comment]
v10.19.0 both in server and local

**Comment by davellanedam** [comment]
I monitored RAM usage when building, it eats ALL of the 1GB of RAM. I just tried resizing the VM to 2GB of RAM and it works now. Any ideas? I'm currently paying 5 Dollars for this and want to keep it like that, if I double the RAM price will rise to 10 dollars.

**Comment by davellanedam** [comment]
UPDATE: After resizing back I tried to build again 2 times and it worked out in both! Looks Like some weird behaviour on the VM. All good now! And it's below 500MB of RAM when building now.

Thanks for the help! I really appreciate it!

**Comment by markim** [comment]
Interesting!! Thanks for the update. I know the front end builds end up using more memory.

I've got a kubernetes cluster with digitalocean that I use to integrate all of my projects. It connects with my digitalocean hosted gitlab server. If you ever need storage / a place to host anything let me know :) All we would do is make a dockerfile for this project, then gitlab/kubernetes hands the deployment and everything on commit.

It's actually pretty cool-- it makes an environment subdomain for each merge request, so you can test out the changes before moving to staging (if wanted), and the production after. Gitlab community edition has it's own mattermost (slack clone) as well, and I just set up the notifications / commands for projects on it.

I sound like a gitlab spokesperson, but I use and like their free version a lot for ci/cd. Gives you a docker registry, and soon an NPM registry as well. Let me know if you want to know more!

Anyways I'm rambling, have a great rest of your weekend! Glad the issue was fixed on vm resizing :D

**Comment by davellanedam** [comment]
Oh that's really nice. I use `git-auto-deploy` and old good bash script. I know how GitLab works, had some projects there and you are right. CD is really nice with them.

It sounds really nice what you are doing there, would love to learn!

Great weekend for you too. And stay safe.

Cheers.

**Pull request merged**